src/repositories/repository.ts

```diff
@@ -0,0 +1,7 @@
+export default interface IRepository<T> {
+    create(data: T): Promise<T>;
+    getAll(): Promise<T[]>;
+    get(id: string): Promise<T>;
+    update(id: string, data: any): Promise<T>;
```

Comment:
SUGGESTION:
to avoid `any` here, you could possibly add a separate model with nullable fields

Reply:
Interesting idea, this interface will be used only in this case I suppose?

Reply:
yes, just a set of fields that possibly can get updated